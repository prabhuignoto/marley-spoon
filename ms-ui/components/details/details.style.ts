import styled from "styled-components";

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 1200px;
  margin: 0 auto;
  box-shadow: 0 0 15px 4px rgba(0,0,0,0.2);
  border-radius: 4px;
  padding: 0 2rem 1rem;
  font-family: Open Sans, Arial, Helvetica, sans-serif;
  margin: 1rem 0;
`;

export const Header = styled.header`
  width: 100%;
  display: flex;
  align-items: center;
  border-bottom: 1px solid #e8e8e8;
  padding: 1rem 0;
  `;

export const Title = styled.h4`
  align-self: flex-start;
  margin-left: 2rem;
  font-size: 1.5rem;
  font-weight: 600;
  padding: 0;
  margin: 0;
`;

export const ImageWrapper = styled.div`
  width: 100%;
  height: 600px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 1rem;
`;

export const Image = styled.img`
  max-width: 100%;
  max-height: 100%;
`;

export const Description = styled.p`
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  width: 90%;
  font-size: 1.2rem;
  padding: 1rem;
  font-weight: 400;
  border-radius: 6px;
  background: #f5f5f5;
`;

export const Chef = styled.div`
  padding: 1rem;
  font-weight: 600;
`
