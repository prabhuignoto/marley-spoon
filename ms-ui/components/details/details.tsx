import React from "react";
import {
  Chef,
  Description,
  Header,
  Image,
  ImageWrapper,
  Title,
  Wrapper,
} from "./details.style";

interface DetailsModel {
  title: string;
  description: string;
  url: string;
  chef: string;
}

const Details: React.FunctionComponent<DetailsModel> = ({
  title,
  description,
  url,
  chef,
}) => {
  return (
    <Wrapper>
      <Header>
        <Title>{title}</Title>
      </Header>
      <ImageWrapper>
        <Image src={url} />
      </ImageWrapper>
      <Description>{description}</Description>
      {chef && <Chef>{`Chef: ${chef}`}</Chef>}
    </Wrapper>
  );
};

export { Details };
