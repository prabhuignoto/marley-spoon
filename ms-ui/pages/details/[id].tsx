import { gql } from "@apollo/client";
import client from "../../apollo-client";
import { Details as DetailsUI } from "../../components/details/details";
import Head from "../../components/common/head";
import { Layout } from "../../components/common/styles";
import { HeaderView } from "../../components/common/header";
import { FooterView } from "../../components/common/footer";

export default function Details({
  id,
  title,
  chefName,
  imageUrl,
  description,
}) {
  return (
    <Layout>
      <Head />
      <HeaderView />
      <DetailsUI
        title={title}
        chef={chefName}
        url={imageUrl}
        description={description}
      />
      <FooterView />
    </Layout>
  );
}

export async function getStaticPaths() {
  const { data } = await client.query({
    query: gql`
      query Recipes {
        getAllRecipes {
          id
        }
      }
    `,
  });

  const paths = data.getAllRecipes.map((recipe) => ({
    params: { id: recipe.id },
  }));

  return {
    paths,
    fallback: false,
  };
}

export async function getStaticProps({ params }) {
  const { data: recipeData } = await client.query({
    query: gql`
      query Recipe($id: String) {
        getRecipe(id: $id) {
          id
          chefId
          tags
          photoId
          description
          title
        }
      }
    `,
    variables: {
      id: params.id,
    },
  });

  const { data } = await client.query({
    query: gql`
      query Query($photoId: String, $chefId: String) {
        getAssetById(id: $photoId) {
          id
          url
          size
        }
        getChefName(id: $chefId)
      }
    `,
    variables: {
      photoId: recipeData.getRecipe.photoId,
      chefId: recipeData.getRecipe.chefId,
    },
  });

  return {
    props: {
      id: recipeData.getRecipe.id,
      title: recipeData.getRecipe.title,
      description: recipeData.getRecipe.description,
      chefName: data.getChefName,
      imageUrl: data.getAssetById.url,
    },
  };
}
