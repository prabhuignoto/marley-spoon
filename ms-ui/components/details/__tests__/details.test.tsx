import { render, screen } from "@testing-library/react";
import { Details } from "../details";

describe("check if snapshots are created", () => {
  it("render snapshots", () => {
    const {container} = render(
      <Details
        title={"title"}
        chef={"dennis"}
        description={"this is a test description"}
        url={"http://dummy.url"}
      />
    );

    expect(container).toMatchSnapshot();
  });
});

describe("check if details rendered correctly", () => {
  it("render snapshots", () => {
    const {container} = render(
      <Details
        title={"title"}
        chef={"dennis"}
        description={"this is a test description"}
        url={"http://dummy.url"}
      />
    );

    expect(screen.getByText("title")).toBeInTheDocument();
    expect(screen.getByText("this is a test description")).toBeInTheDocument();
    expect(screen.getByText("this is a test description")).toBeInTheDocument();
  });
});
