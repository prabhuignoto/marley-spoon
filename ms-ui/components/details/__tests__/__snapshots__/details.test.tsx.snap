// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if snapshots are created render snapshots 1`] = `
<div>
  <div
    class="sc-bdnxRM bYAabF"
  >
    <header
      class="sc-gtsrHT kYGifz"
    >
      <h4
        class="sc-dlnjwi ctTxXO"
      >
        title
      </h4>
    </header>
    <div
      class="sc-hKFxyN ePMirl"
    >
      <img
        class="sc-eCApnc iiYCoF"
        src="http://dummy.url"
      />
    </div>
    <p
      class="sc-jSFjdj ipSlzg"
    >
      this is a test description
    </p>
    <div
      class="sc-gKAaRy bZyPSH"
    >
      Chef: dennis
    </div>
  </div>
</div>
`;
